src/nft/components/bag/BagFooter.tsx

```diff
@@ -377,105 +379,78 @@ export const BagFooter = ({ setModalIsOpen, eventProperties }: BagFooterProps) =
     helperTextColor,
     handleClick,
     buttonColor,
-  } = useMemo(() => {
-    let handleClick: (() => void) | (() => Promise<void>) = fetchAssets
-    let buttonText = <Trans>Something went wrong</Trans>
-    let disabled = true
-    let warningText = undefined
-    let warningTextColor = theme.accentWarning
-    let helperText = undefined
-    let helperTextColor = theme.textSecondary
-    let buttonColor = theme.accentAction
-    let buttonTextColor = theme.accentTextLightPrimary
-
+  } = useMemo((): BuyButtonState => {
     if (connected && chainId !== SupportedChainId.MAINNET) {
```

Comment:
Thoughts on these multiple return calls vs instantiating lets for the arguments for `getBuyButtonStateData`, updating based on if statements, and having 1 return fn call at the end?

Reply:
the point of this refactor was to have clear data-in -> data-out, having a mutable object makes that less clear and its only/should going to get mutated once. Also this doesn't require you have to go down all of the branches to return